internal/tiltfile/tiltfile_state.go

```diff
@@ -1068,8 +1056,20 @@ func (s *tiltfileState) imgTargetsForDependencyIDsHelper(ids []model.TargetID, c
 		}
 		claimStatus[id] = claimPending
 
+		registry := s.defaultReg
+		if s.orchestrator() == model.OrchestratorK8s && !s.privateRegistry.Empty() {
+			// If we've found a private registry in the cluster at run-time,
+			// use that instead of the one in the tiltfile
+			s.logger.Infof("Auto-detected private registry from environment: %s", s.privateRegistry)
+			registry = s.privateRegistry
+		}
+		refs, err := container.NewRefSet(image.configurationRef, registry)
```

Comment:
would it make more sense to do this substitution during tiltfile execution? So RefSet would go on docker_image.

The problem with this approach is that if there's an error, you'll lose the stack trace, as in this bug: https://github.com/windmilleng/tilt/issues/2768

so we try hard not to return errors from this function without lots of context about what went wrong and how to fix it.